lib/polyn/serializers/json.rb

```diff
@@ -24,13 +24,45 @@ module Serializers
     class Json
       def self.serialize!(nats, event, **opts)
         validate_event_instance!(event)
-        validate!(nats, event.to_h, **opts)
+        event = event.to_h
+        validate!(nats, event, **opts)
+        JSON.generate(event)
+      end
+
+      def self.deserialize!(nats, json, **opts)
+        data = deserialize(nats, json, **opts)
+        raise data if data.is_a?(Polyn::Errors::Error)
+
+        data
+      end
+
+      def self.deserialize(nats, json, **opts)
+        data = decode(json)
+        return data if data.is_a?(Polyn::Errors::Error)
+
+        error = validate(nats, data, **opts)
+        return error if error.is_a?(Polyn::Errors::Error)
+
+        data   = Polyn::Utils::Hash.deep_symbolize_keys(data)
+        Event.new(data)
+      end
+
+      def self.decode(json)
+        JSON.parse(json)
```

Comment:
It would save us a full recursive step later if we'd pass in the option `symbolize_names: true` so it just symbolizes the keys on the way out of the JSON parse call.